pkg/stub/handler.go

```diff
@@ -104,6 +107,99 @@ type Handler struct {
 	mutex *sync.Mutex
 }
 
+func (h *Handler) StatusUpdate(condition v1alpha1.SamplesResourceConditionType, srcfg *v1alpha1.SamplesResource) error {
+	if condition == v1alpha1.SamplesExist {
+		cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "unable to get configm map: %v")
+		}
+		if cm == nil {
+			cm := corev1.ConfigMap{}
+			cm.Name = v1alpha1.SamplesResourceName
+			cm.Data = map[string]string{}
+			if len(srcfg.Spec.InstallType) == 0 {
+				cm.Data[installtypekey] = string(v1alpha1.CentosSamplesDistribution)
+			} else {
+				cm.Data[installtypekey] = string(srcfg.Spec.InstallType)
+			}
+			if len(srcfg.Spec.Architectures) == 0 {
+				cm.Data[x86] = x86
+			} else {
+				for _, arch := range srcfg.Spec.Architectures {
+					switch arch {
+					case x86:
+						cm.Data[x86] = x86
+					case ppc:
+						cm.Data[ppc] = ppc
+					}
+				}
+			}
+			_, err = h.configmapclientwrapper.Create(h.namespace, &cm)
+			if err != nil {
+				return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "problem creating configmap: %v")
+			}
+			return nil
+		}
+		err = fmt.Errorf("should never update status config map after initial creation: %#v", srcfg)
+		return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "%v")
+	}
+	return nil
+}
+
+func (h *Handler) SpecValidation(srcfg *v1alpha1.SamplesResource) error {
+	// if we have not had a valid SamplesResource processed, allow caller to try with
+	// the srcfg contents
+	if h.samplesResource == nil || !h.samplesResource.ConditionTrue(v1alpha1.SamplesExist) {
+		return nil
+	}
+	cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
+	if cm == nil || err != nil {
+		return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "problem getting configmap: %v")
+	}
+
+	switch srcfg.Spec.InstallType {
+	case "":
+		fallthrough
```

Comment:
what's the intent here?

Reply:
safety precaution ... in case this method is ever called with InstallType empty, we employ the defaulting convention we have for an empty samplesresource

Reply:
`case "",v1alpha1.CentosSamplesDistribution:` is a cleaner way to write this imho

fallthrough makes more sense when you want to do some logic in this case statement *and* execute the logic in the next case statement.